fearless-utils/src/main/java/jp/co/soramitsu/fearless_utils/runtime/definitions/v14/TypesParserV14.kt

```diff
@@ -0,0 +1,184 @@
+                        )
+                    }
+                    def.schema is TypeDefSequence -> {
+                        Vec(
+                            name.toString(),
+                            resolveTypeAllWaysOrCreate(params, def[TypeDefSequence.type].toString())
+                        )
+                    }
+                    def.schema is TypeDefVariant -> {
+                        val list = def[TypeDefVariant.variants]
+                        val res = list.map {
+                            val ch = it[TypeDefVariantItem.fields2]
+                            val itemName = it[TypeDefVariantItem.index].toString()
+                            val children = parseTypeMapping(params, ch as List<*>)
+                            val s = Struct(itemName, children)
+                            DictEnum.Entry(it[TypeDefVariantItem.name], TypeReference(s))
+                        }
+                        DictEnum(name.toString(), res)
+                    }
+                    def.schema is TypeDefCompact -> {
+                        Compact(name.toString())
+                    }
+                    def.schema is TypeDefBitSequence -> {
+                        Tuple(
+                            name.toString(),
+                            listOf(
+                                def[TypeDefBitSequence.bit_store_type],
+                                def[TypeDefBitSequence.bit_order_type]
+                            ).map { resolveTypeAllWaysOrCreate(params, it.toString()) }
+                        )
+                    }
+                    else -> {
+                        null
+                    }
+                }
+            }
+            is TypeDefEnum -> {
+                when (def) {
+                    TypeDefEnum.str -> {
+                        Alias(name.toString(), TypeReference(Bytes))
+                    }
+                    TypeDefEnum.char -> {
+                        Alias(name.toString(), TypeReference(Bytes))
+                    }
+                    else -> {
+                        Alias(name.toString(), typesBuilder.getOrCreate(def.localName))
+                    }
+                }
+            }
+            is List<*> -> {
+                (def as? List<BigInteger>)?.let { list ->
+                    Tuple(
+                        name.toString(),
+                        list.map { resolveTypeAllWaysOrCreate(params, it.toString()) }
+                    )
+                }
+            }
+            else -> {
+                null
+            }
+        }
+    }
+
+    private fun parseTypeMapping(
+        params: Params,
+        struct: List<*>,
+    ): LinkedHashMap<String, TypeReference> {
+        val children = LinkedHashMap<String, TypeReference>()
+        for (type in struct) {
+            val s = type as EncodableStruct<*>
+            when {
+                s.schema is TypeDefCompositeField -> {
```

Comment:
It looks like typing can be imrpoved here since `struct` is always `List<EncodableStruct<TypeDefCompositeField >>`